packages/communication-react/src/mergedHooks.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import {
+  DefaultCallingHandlers,
+  getCallingSelector,
+  GetCallingSelector,
+  useCallingHandlers,
+  useCallingSelector
+} from 'calling-component-bindings';
+import {
+  DefaultChatHandlers,
+  getChatSelector,
+  GetChatSelector,
+  useChatHandlers,
+  useChatSelector
+} from 'chat-component-bindings';
+import { ChatClientState } from 'chat-stateful-client';
+import { CallClientState } from 'calling-stateful-client';
+import { Common } from 'acs-ui-common';
+
+export type ClientState = CallClientState & ChatClientState;
+export type Selector = (state: ClientState, props: any) => any;
+
+// Because of react hooks rules, hooks can't be conditionally called
+// We call both call and chat hooks and detect current context
+// Return undefined and skip execution when not in that context
+export const useSelector = <ParamT extends Selector | undefined>(
+  selector: ParamT,
+  selectorProps?: ParamT extends Selector ? Parameters<ParamT>[1] : undefined
+): ParamT extends Selector ? ReturnType<ParamT> : undefined => {
+  const callProps = useCallingSelector(selector as any, selectorProps);
+  const chatProps = useChatSelector(selector as any, selectorProps);
```

Comment:
Do we need to document that developers shouldn't nest Call and Chat providers to avoid unexpected behavior?

Reply:
Users can do it, this hook will detect which context it is under. Just one scenario they need to choose which context on in usePropsFor, most likely it is fine(because all of the components are designed for only call/chat, except participant list)